src/bin/pg_autoctl/cli_show.c

```diff
@@ -528,3 +563,354 @@ cli_show_monitor_uri(int argc, char **argv)
+				}
+				fileOptions.selection = SHOW_FILE_STATE;
+				log_trace("--state");
+				break;
+			}
+
+			case 'i':
+			{
+				if (fileOptions.selection != SHOW_FILE_UNKNOWN
+					&& fileOptions.selection != SHOW_FILE_INIT)
+				{
+					log_error(
+						"Please use only one of --config --state --init --pid");
+					commandline_help(stderr);
+				}
+				fileOptions.selection = SHOW_FILE_INIT;
+				log_trace("--init");
+				break;
+			}
+
+			case 'p':
+			{
+				if (fileOptions.selection != SHOW_FILE_UNKNOWN
+					&& fileOptions.selection != SHOW_FILE_PID)
+				{
+					log_error(
+						"Please use only one of --config --state --init --pid");
+					commandline_help(stderr);
+				}
+				fileOptions.selection = SHOW_FILE_PID;
+				log_trace("--pid");
+				break;
+			}
+
+			case 'V':
+			{
+				/* keeper_cli_print_version prints version and exits. */
+				keeper_cli_print_version(argc, argv);
+				break;
+			}
+
+			case 'v':
+			{
+				++verboseCount;
+				switch (verboseCount)
+				{
+					case 1:
+						log_set_level(LOG_INFO);
+						break;
+
+					case 2:
+						log_set_level(LOG_DEBUG);
+						break;
+
+					default:
+						log_set_level(LOG_TRACE);
+						break;
+				}
+				break;
+			}
+
+			case 'q':
+			{
+				log_set_level(LOG_ERROR);
+				break;
+			}
+
+			case 'h':
+			{
+				commandline_help(stderr);
+				exit(EXIT_CODE_QUIT);
+				break;
+			}
+
+			default:
+			{
+				log_error("Failed to parse command line, see above for details.");
+				commandline_help(stderr);
+				exit(EXIT_CODE_BAD_ARGS);
+				break;
+			}
+		}
+	}
+
+	/* default to --all when no option has been selected */
+	if (fileOptions.selection == SHOW_FILE_UNKNOWN)
+	{
+		fileOptions.selection = SHOW_FILE_ALL;
+	}
+
+	keeperOptions = options;
+	showFileOptions = fileOptions;
+
+	return optind;
+}
+
+
+/*
+ * cli_show_files lists the files used by pg_autoctl.
+ */
+static void
+cli_show_file(int argc, char **argv)
+{
+	KeeperConfig config = keeperOptions;
+	pgAutoCtlNodeRole role = PG_AUTOCTL_ROLE_UNKNOWN;
+
+	if (!keeper_config_set_pathnames_from_pgdata(&config.pathnames,
+												 config.pgSetup.pgdata))
+	{
+		/* errors have already been logged */
+		exit(EXIT_CODE_BAD_CONFIG);
+	}
+
+	role = ProbeConfigurationFileRole(config.pathnames.config);
+
+	switch (showFileOptions.selection)
+	{
+		case SHOW_FILE_ALL:
+		{
+			fprintf(stdout, "config: \"%s\"\n", config.pathnames.config);
+
+			if (role == PG_AUTOCTL_ROLE_KEEPER)
+			{
+				fprintf(stdout, " state: \"%s\"\n", config.pathnames.state);
+				fprintf(stdout, "  init: \"%s\"\n", config.pathnames.init);
+				fprintf(stdout, "   pid: \"%s\"\n", config.pathnames.pid);
+			}
+			break;
+		}
+
+		case SHOW_FILE_CONFIG:
+		{
+			if (showFileOptions.showFileContents)
+			{
+				if (!fprint_file_contents(config.pathnames.config))
+				{
+					/* errors have already been logged */
+					exit(EXIT_CODE_BAD_CONFIG);
+				}
+			}
+			else
+			{
+				fprintf(stdout, "\"%s\"\n", config.pathnames.config);
```

Comment:
the extra quotes seem tedious for parsing the output.

Reply:
I think it's needed when file names contain spaces and sometimes other characters too, so I'd rather have extra quoting always, just in case? Alternative is to actually quote and escape only when needed... is there a good way to do the escaping in a portable way? I figured always double-quoting is cheap and works almost always...

Reply:
Why is it needed? If I wish to parse the output, can't I just read everything up to the end of line?

Reply:
Good point, done that way now, with some more changes. Thinking more about it, now that we have JSON support internally and that we emit more of it, I think it would be a good format for the `--all` output. And `jq` makes it very easy to then use that format if you want to script around it.

```
$ pg_autoctl show file --pgdata /tmp/sm/x
{
    "config": "\/Users\/dim\/.config\/pg_autoctl\/tmp\/sm\/x\/pg_autoctl.cfg",
    "state": "\/Users\/dim\/.local\/share\/pg_autoctl\/tmp\/sm\/x\/pg_autoctl.state",
    "init": "\/Users\/dim\/.local\/share\/pg_autoctl\/tmp\/sm\/x\/pg_autoctl.init",
    "pid": "\/tmp\/pg_autoctl\/tmp\/sm\/x\/pg_autoctl.pid"
}

$ pg_autoctl show file --pgdata /tmp/sm/m
{
    "config": "\/Users\/dim\/.config\/pg_autoctl\/tmp\/sm\/m\/pg_autoctl.cfg"
}

$ pg_autoctl show file --pgdata /tmp/sm/m | jq .
{
  "config": "/Users/dim/.config/pg_autoctl/tmp/sm/m/pg_autoctl.cfg"
}

$ pg_autoctl show file --pgdata /tmp/sm/m | jq .config
"/Users/dim/.config/pg_autoctl/tmp/sm/m/pg_autoctl.cfg"

$ pg_autoctl show file --pgdata /tmp/sm/x | jq .config
"/Users/dim/.config/pg_autoctl/tmp/sm/x/pg_autoctl.cfg"

$ pg_autoctl show file --pgdata /tmp/sm/x --config
/Users/dim/.config/pg_autoctl/tmp/sm/x/pg_autoctl.cfg
```